tempesta_fw/t/unit/test_http_parser.c

```diff
@@ -85,45 +165,64 @@ reset_req(size_t data_len)
  *  >  0 - EOF: all possible fragments are parsed, terminate the loop.
  */
 int
-do_split_and_parse(unsigned char *req_str)
+do_split_and_parse(unsigned char *str, int type)
 {
 	static char head_buf[PAGE_SIZE];
 	static char tail_buf[PAGE_SIZE];
-	static size_t req_len, head_len, tail_len;
+	static size_t len, head_len, tail_len;
```

Comment:
Well, we define the variables static to make a loop. This is not the better and easier to understand solution. The function should be reworked as set of loops and functions